src/pages/Environments.js

```diff
@@ -202,48 +206,51 @@ const Environments = () => {
-    return h(Fragment, [
-      app.googleProject,
-      inactive && h(TooltipTrigger, {
-        content: 'This billing project has multiple active cloud environments. Only the most recently created one will be used.'
-      }, [icon('warning-standard', { style: { marginLeft: '0.25rem', color: colors.warning() } })])
-    ])
-  }
+  // We start the first output string with an empty space because empty space would
+  // not apply to the case where appType is not defined (e.g. Jupyter, RStudio).
+  const forAppText = appType => !!appType ? ` for ${_.capitalize(appType)}` : ''
 
-  const renderBillingProjectRuntime = runtime => {
-    const inactive = !_.includes(runtime.status, ['Deleting', 'Error']) &&
-      getCurrentRuntime(runtimesByProject[runtime.googleProject]) !== runtime
+  const getWorkspaceCell = (namespace, name, appType, shouldWarn) => {
     return h(Fragment, [
-      runtime.googleProject,
-      inactive && h(TooltipTrigger, {
-        content: 'This billing project has multiple active cloud environments. Only the most recently created one will be used.'
+      h(Link, { onClick: () => Nav.goToPath('workspace-notebooks', { namespace, name }) }, [name]),
+      shouldWarn && h(TooltipTrigger, {
+        content: `This workspace has multiple active cloud environments${forAppText(appType)}. Only the latest one will be used.`
       }, [icon('warning-standard', { style: { marginLeft: '0.25rem', color: colors.warning() } })])
     ])
   }
+  const renderWorkspaceForApps = app => {
+    const { status, appType, googleProject, labels: { saturnWorkspaceNamespace, saturnWorkspaceName } } = app
+    const shouldWarn = !_.includes(status, ['DELETING', 'ERROR', 'PREDELETING']) &&
+      getCurrentApp(appType)(appsByProject[googleProject]) !== app
+    return getWorkspaceCell(saturnWorkspaceNamespace, saturnWorkspaceName, appType, shouldWarn)
+  }
+  const renderWorkspaceForRuntimes = runtime => {
+    const { status, googleProject, labels: { saturnWorkspaceNamespace, saturnWorkspaceName } } = runtime
+    const shouldWarn = !_.includes(status, ['Deleting', 'Error']) &&
+      getCurrentRuntime(runtimesByProject[googleProject]) !== runtime
+    return getWorkspaceCell(saturnWorkspaceNamespace, saturnWorkspaceName, null, shouldWarn)
+  }
 
-  const renderDetailsApp = (app, disks) => {
-    const { appName, diskName } = app
-    const disk = _.find({ name: diskName }, disks)
+  const getDetailsPopup = (cloudEnvName, googleProject, disk) => {
     return h(PopupTrigger, {
       content: div({ style: { padding: '0.5rem' } }, [
-        div([span({ style: { fontWeight: '600' } }, ['Name: ']), appName]),
-        disk && div([span({ style: { fontWeight: '600' } }, ['Persistent Disk: ']), disk.name])
+        div([strong(['Name: ']), cloudEnvName]),
+        div([strong(['Google Project: ']), googleProject]),
+        disk && div([strong(['Persistent Disk: ']), disk.name])
```

Comment:
I personally prefer this syntax over `!!disk &&…`, but I know you have been trying to move us in the direction of using the !!.

Reply:
Should this be `disk` or  `disk?.name`
I prefer the boolean coercion as it will prevent accidental rendering of some falsey values

Reply:
Thanks. Changing for consistency.

Reply:
I think it should be `disk` because disks should always have names.